### ISSUE TITLE
batch size scheduler

### ISSUE THREAD

**Issue by ghost** [pytorch/ignite#454]
Does the module implement the batch_size scheduler?

**Comment by vfdev-5** [comment]
@EvilPsyCHo no it is not implemented. I think this should be done on `BatchSampler`-side of the `DataLoader`. Thoughts ?

**Comment by aniketpandey05** [comment]
Hi, @vfdev-5 , is this still relevant?  I Checked that  ignite/handlers/ has no BatchSampler based scheduling, and ReproducibleBatchSampler in engine/deterministic.py only handles determinism

**Comment by vfdev-5** [comment]
@rogueslasher what would you suggest to implement? I'm not sure if ignite should have these things

**Comment by aniketpandey05** [comment]
@vfdev-5 A  handler similar to ParamScheduler  takes a dataloader and a scheduling function, updates dataloader.batch_sampler.batch_size on each epoch. would be Useful for curriculum learning where you want to gradually increase batch size during training

**Comment by vfdev-5** [comment]
Well, we can do that. This should not be very complicated to code.

**Comment by aniketpandey05** [comment]
Hey @vfdev-5 , i have  Added BatchSizeScheduler extending BaseParamScheduler . Also added tests for asserts, save_history, simulate_values and state_dict.
